app/components/Reactions/ViewReactionsDialog.tsx

```diff
@@ -0,0 +1,148 @@
+import Text from "~/components/Text";
+import { hover } from "~/styles";
+import { EmojiReactedUsers, ReactionData } from "~/types";
+
+type Props = {
+  /** Callback to fetch the detailed reaction data (incl. users' info) */
+  fetchReactionData: () => Promise<ReactionData[]>;
+};
+
+const ViewReactionsDialog: React.FC<Props> = ({ fetchReactionData }) => {
+  const { t } = useTranslation();
+  const tab = useTabState();
+
+  const [emojiReactedUsers, setEmojiReactedUsers] =
+    React.useState<EmojiReactedUsers>({});
+
+  React.useEffect(() => {
+    const fetchReactions = async () => {
+      try {
+        const reactionData = await fetchReactionData();
+
+        const transformedData = reactionData.reduce((acc, data) => {
+          const emoji = data.emoji;
+          const users = (acc[emoji] ?? []) as EmojiReactedUsers[number];
+          users.push(data.user);
+          acc[emoji] = users;
+          return acc;
+        }, {} as EmojiReactedUsers);
+
+        setEmojiReactedUsers(transformedData);
+      } catch (err) {
+        toast.error(t("Could not load reactions"));
+      }
+    };
+
+    void fetchReactions();
+  }, [t, setEmojiReactedUsers, fetchReactionData]);
+
+  if (!emojiReactedUsers) {
+    return <PlaceHolder />;
+  }
+
+  return (
+    <>
+      <TabActionsWrapper>
+        {Object.keys(emojiReactedUsers).map((emoji) => (
+          <StyledTab
+            {...tab}
+            key={emoji}
+            id={emoji}
+            aria-label={t("Reaction")}
+            $active={tab.selectedId === emoji}
+          >
+            <Emoji size={16}>{emoji}</Emoji>
+          </StyledTab>
+        ))}
+      </TabActionsWrapper>
+      {Object.entries(emojiReactedUsers).map(([emoji, users]) => (
+        <StyledTabPanel {...tab} key={emoji}>
+          {users.map((user) => (
+            <UserInfo key={user.name} align="center" gap={8}>
+              <Avatar model={user} size={AvatarSize.Medium} />
+              <Text size="medium">{user.name}</Text>
+            </UserInfo>
+          ))}
+        </StyledTabPanel>
+      ))}
+    </>
+  );
+};
+
+const PlaceHolder = React.memo(
+  () => (
+    <>
+      <TabActionsWrapper gap={8} style={{ paddingBottom: "10px" }}>
+        <PlaceholderText width={40} height={32} />
+        <PlaceholderText width={40} height={32} />
+      </TabActionsWrapper>
+      <UserInfo align="center" gap={12}>
+        <PlaceholderText width={AvatarSize.Medium} height={AvatarSize.Medium} />
+        <PlaceholderText height={34} />
+      </UserInfo>
+      <UserInfo align="center" gap={12}>
+        <PlaceholderText width={AvatarSize.Medium} height={AvatarSize.Medium} />
+        <PlaceholderText height={34} />
+      </UserInfo>
+    </>
+  ),
+  () => true
+);
+PlaceHolder.displayName = "ViewReactionsPlaceholder";
+
+const TabActionsWrapper = styled(Flex)`
+  border-bottom: 1px solid ${s("inputBorder")};
+`;
+
+const StyledTab = styled(Tab)<{ $active: boolean }>`
+  position: relative;
```

Comment:
This is basically completely restyled, is it neccessary?

Reply:
This unsets the `reakit` tab styles with `Outline` styles.